client/commonFramework/end.js

```diff
@@ -112,6 +112,7 @@ $(document).ready(function() {
         common.displayTestResults(tests);
         if (solved) {
           common.showCompletion();
+          common.codeStorage.removeStoredValue(common.challengeName);
```

Comment:
With at least the new code we're adding we can spot such trivial things and make it a little better. About _

> from variables that don't need it

I  don't think refactoring the entire file would be needed or expected in the commit. Nevertheless, great work.